### PR TITLE
fix: when creating new user, set email handle as username

### LINKED ISSUE #11206: [CAL-2417] Inviting a member to an org doesn't set username
### Issue Summary

Inviting someone doesn't set the username (which should be email handle) on the created user in the Users table. The username is forced to be email handle upon accepting the invite.

### Steps to Reproduce

1. Create org
2. Invite someone to the org

### Actual Results

Added member shows "No username"

### Expected Results

Added member should show their email handle

<sub>[CAL-2417](https://linear.app/calcom/issue/CAL-2417/inviting-a-member-to-an-org-doesnt-set-username)</sub>

### LINKED ISSUE #11240: [CAL-2429] External invited org user could encounter username clash
### Issue Summary

In the edge case of inviting an external user john@foobar.com on an Org with domain acme.com which has already a john@acme.com, then the external user will not be able to accept the invitation due to username clash and not able to pick an alternative.

### Steps to Reproduce

1. Create the org acme as john@acme.com (`http://app.cal.local:3000/settings/organizations/new`)
2. Verify the org (Admin user at `http://app.cal.local:3000/settings/admin/organizations`)
3. Invite someone from acme, such as peter@acme.com
4. Have the invited user in last step accept the invitation
5. Invite another person this time external one with the same email address handle such as peter@foobar.com
6. Have the invited user in last step accept the invitation

### Actual Results

User peter@foobar.com will not be able to accept the invitation receiving an error regarding username `peter` being already in use, with no chance to select an alternative username (as it is inferred and disabled)

### Expected Results

Always set username for external users as email address handle hyphen the email domain name, such as peter-foobar in previous example.

<sub>[CAL-2429](https://linear.app/calcom/issue/CAL-2429/external-invited-org-user-could-encounter-username-clash)</sub>

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

This PR sets new org member user's usernames as their email handles. When new users are signing up to Cal.com we already set their username on the signup form as their email handle.

CleanShot 2023-09-08 at 12 41 50

Fixes #11206 
Fixes #11240

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- In an org invite a new member
- On the member page, it should show their email handle as their username
- Confirm this in the DB for the new user created as well

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by leog — CHANGES_REQUESTED** [review]
LGTM, although there can be a conflict with external users, it would be great if [this other ticket](https://github.com/calcom/cal.com/issues/11240) can be addressed in this PR

**Comment by joeauyeung** [comment]
> LGTM, although there can be a conflict with external users, it would be great if [this other ticket](https://github.com/calcom/cal.com/issues/11240) can be addressed in this PR

Ah I knew this issue was too easy. I'll go ahead and tackle the issue you linked as well.

**Comment by PeerRich** [comment]
this is awesome

**Review by leog — APPROVED** [review]
Changes made, thanks @joeauyeung for starting this PR

File: packages/trpc/server/routers/viewer/teams/inviteMember/utils.ts [hunk 1]
```diff
@@ -136,17 +137,26 @@ export async function createNewUserConnectToOrgIfExists({
...
+  autoAcceptEmailDomain?: string;
   connectionInfo: ReturnType<typeof getOrgConnectionInfo>;
 }) {
   const { orgId, autoAccept } = connectionInfo;
 
+  const [emailUser, emailDomain] = usernameOrEmail.split("@");
+  const username =
+    emailDomain === autoAcceptEmailDomain
+      ? slugify(emailUser)
+      : slugify(`${emailUser}-${emailDomain.split(".")[0]}`);
```
Comment (zomars) [inline packages/trpc/server/routers/viewer/teams/inviteMember/utils.ts:155]: This is causing a weird edge case where there might be the possibility that the generated user may exist already. causing a prisma error. Example: we have `robert@example.com` being converted to `robert-example` username. Which already exists in prod. On production this only returns a 500 error:

On dev we get this:

```
Object { message: "\nInvalid `prisma.user.create()` invocation:\n\n\nUnique constraint failed on the fields: (`username,`)", code: -32603, data: {…} }
```

**Pull request merged**